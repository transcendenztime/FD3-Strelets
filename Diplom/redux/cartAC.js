const QTY_ADD = 'QTY_ADD';
const PROD_ADD = 'PROD_ADD';
const PROD_DEL = 'PROD_DEL';
const SUM_QTY = 'SUM_QTY';
const QTY_TO_NULL = 'QTY_TO_NULL';
const DELIVERY_CHNG = 'DELIVERY_CHNG';

const CART_CLEAR = 'CART_CLEAR';

const prod_add = function(prodid, info) {
  return {
    type: PROD_ADD,
    prodid: prodid,
    info: info,
  };
}

const prod_del = function(prodid) {
  return {
    type: PROD_DEL,
    prodid: prodid,
  };
}

const cart_clear = function() {
  return {
    type: CART_CLEAR,
  };
}

const qty_add = function(prodid, newQTY, newSum) {
  return {
    type: QTY_ADD,
    prodid: prodid,
    newQTY: newQTY,
    newSum: newSum
  };
}

const sum_qty = function(qty) {
  return {
    type: SUM_QTY,
    qty: qty,
  }
}

const qty_to_null = function() {
  return {
    type: QTY_TO_NULL,
  }
}

const delivery_chng = function(num) {
  return {
    type: DELIVERY_CHNG,
    num: num,
  }
}

export {
  prod_add, PROD_ADD,
  prod_del, PROD_DEL,
  cart_clear, CART_CLEAR,
  qty_add, QTY_ADD,
  sum_qty, SUM_QTY,
  qty_to_null, QTY_TO_NULL,
  delivery_chng, DELIVERY_CHNG,
}
