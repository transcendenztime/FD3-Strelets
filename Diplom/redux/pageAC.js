const PAGE_CHANGE = 'PAGE_CHANGE';

const page_change = function(page) {
  return {
    type: PAGE_CHANGE,
    page: page,
  };
}

export {
  page_change, PAGE_CHANGE,
}
