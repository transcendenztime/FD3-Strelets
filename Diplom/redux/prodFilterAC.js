const PROD_FILTER = 'PROD_FILTER';

const prod_filter = function(name) {
  return {
    type: PROD_FILTER,
    name,
  };
}

export {
    prod_filter, PROD_FILTER,
}