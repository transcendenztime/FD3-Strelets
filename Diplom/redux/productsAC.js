const PRODUCTS_LOADING='PRODUCTS_LOADING';
const PRODUCTS_ERROR='PRODUCTS_ERROR';
const PRODUCTS_SET='PRODUCTS_SET';

const productsLoadingAC = function() {
  return {
    type: PRODUCTS_LOADING,
  };
}

const productsErrorAC = function() {
  return {
    type: PRODUCTS_ERROR,
  };
}

const productsSetAC = function( products ) {
  return {
    type: PRODUCTS_SET,
    products: products,
  };
}

export {
  productsLoadingAC, PRODUCTS_LOADING,
  productsErrorAC, PRODUCTS_ERROR,
  productsSetAC, PRODUCTS_SET,
}
