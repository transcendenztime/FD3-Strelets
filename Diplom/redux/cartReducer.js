import {  PROD_ADD, PROD_DEL, CART_CLEAR, QTY_ADD, DELIVERY_CHNG } from './cartAC';

const initState={

  // ключ - идентификатор prod, значение - info
  products: {},
  delivery: 0,  
}


function cartReducer(state = initState, action) {
  switch (action.type) {
    
    case PROD_ADD: {
      let newProdInfo = action.info;
      
      //если в корзине уже есть данный товар, суммируем добавляемое с тем, что есть
      if (state.products[action.prodid]) {
        newProdInfo.qty += state.products[action.prodid].qty;
        newProdInfo.sum += state.products[action.prodid].sum;
      }

      let newState = {...state,
        products: {...state.products,
          [action.prodid]: newProdInfo}};
      return newState;
    }    

    case PROD_DEL: {
      let newState = {...state};
      delete newState.products[action.prodid];
      return newState;
    }    

    case CART_CLEAR: {
      //чистим корзину
      let newState = {...initState};
      return newState;
    }

    case QTY_ADD: {
      //меняем количество единиц товара и общую сумму денег за это количество единиц товара 
      let newState = {...state,
        products: {...state.products,
          [action.prodid]: {...state.products[action.prodid],
            qty: action.newQTY, sum: action.newSum }
        }
      };

      return newState;
    }
    
    case DELIVERY_CHNG: {
      let newState = {...state,
        delivery: action.num};
      return newState;
    } 
    
    default:
      return state;
  }
}

export default cartReducer;
