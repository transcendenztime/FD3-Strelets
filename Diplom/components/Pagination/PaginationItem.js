"use strict";

import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import {connect} from 'react-redux';
import { page_change } from "../../redux/pageAC";

class PaginationItem extends React.Component {
  
  static propTypes = { 
    num: PropTypes.number.isRequired, //передан из родительского компонента
    startLink: PropTypes.string.isRequired, //передан из родительского компонента
    page: PropTypes.object.isRequired,
  };


  changePage = () => {
    this.props.dispatch( page_change(this.props.num) );
  }
  
  render() {
    
    return (
        <li className = "pagination-item">
          <NavLink to = {this.props.num != 1 ? 
              this.props.startLink + '/' + this.props.num + "page" : this.props.startLink} onClick = {this.changePage}
              className = {this.props.num == this.props.page.page ? "activePage" : null}>
            {this.props.num}
          </NavLink>
        </li>          
    );

  }

}
   
const mapStateToProps = function (state) {
    return {    
      page: state.page,
    };
  };
  
  export default connect(mapStateToProps)(PaginationItem);