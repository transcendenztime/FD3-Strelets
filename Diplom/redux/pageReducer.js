import {  PAGE_CHANGE } from './pageAC';

const initState={
    page: 1,
}

function pageReducer(state = initState, action) {
  switch (action.type) {
    
    case PAGE_CHANGE: {
      
      //если страница та же, ничего не делаем
      if (state.page != action.page) {
        let newState = {...state,
          page: action.page};
        return newState;
      } else {return state}
      
    }    
    
    default:
      return state;
  }
}

export default pageReducer;
