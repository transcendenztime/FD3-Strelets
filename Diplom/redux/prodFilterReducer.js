import {  PROD_FILTER } from './prodFilterAC';

const initState={
    prodFilter: '',
}


function prodFilterReducer(state = initState, action) {
  switch (action.type) {
    
    case PROD_FILTER: {
            
      let newState = {...state,
        prodFilter: action.name};
        console.log(action.name)
      return newState;
      
    }    
    
    default:
      return state;
  }
}

export default prodFilterReducer;
