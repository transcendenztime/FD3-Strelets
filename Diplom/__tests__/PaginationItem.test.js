"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import { createStore } from 'redux';
import combinedReducer from '../redux/reducers.js';
import { Provider } from 'react-redux';

import PaginationItem from '../components/Pagination/PaginationItem';
  

let store=createStore(combinedReducer);

test('внешний вид и работа PaginationItem', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
    <Router>
      <Provider store={store}>
        <PaginationItem num = {1} startLink ={'/catalogue'}/>
      </Provider>
    </Router>
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  // создаём 2 тестовую версию компонента
  const component2 = renderer.create(
    <Router>
      <Provider store={store}>
        <PaginationItem num = {5} startLink ={'/catalogue/category'}/>
      </Provider>
    </Router>
  );

  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree2=component2.toJSON();
  expect(componentTree2).toMatchSnapshot();
        
});
