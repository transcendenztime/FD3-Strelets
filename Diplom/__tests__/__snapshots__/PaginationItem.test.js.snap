// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`внешний вид и работа PaginationItem 1`] = `
<li
  className="pagination-item"
>
  <a
    aria-current={null}
    className="activePage"
    href="/catalogue"
    onClick={[Function]}
    style={undefined}
  >
    1
  </a>
</li>
`;

exports[`внешний вид и работа PaginationItem 2`] = `
<li
  className="pagination-item"
>
  <a
    aria-current={null}
    className={null}
    href="/catalogue/category/5page"
    onClick={[Function]}
    style={undefined}
  >
    5
  </a>
</li>
`;
